symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigLoader.java

```diff
@@ -0,0 +1,156 @@
+package com.symphony.bdk.core.config;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import com.symphony.bdk.core.config.legacy.model.LegacySymConfig;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Slf4j
+public class BdkConfigLoader {
+
+    private static final JsonMapper JSON_MAPPER = new JsonMapper();
+    private static final YAMLMapper YAML_MAPPER = new YAMLMapper();
+
+    /**
+     * Load BdkConfig from a file path
+     *
+     * @param configPath Path of the config file
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromFile(String configPath) {
+        try {
+            return JSON_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
+
+        try {
+            return YAML_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in Yaml format");
+        }
+
+        log.error("Config is not in a valid format");
+        return null;
+    }
+
+    /**
+     * Load BdkConfig from an InputStream
+     *
+     * @param inputStream InputStream
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromInputStream(InputStream inputStream) {
+        try {
+            return JSON_MAPPER.readValue(inputStream, BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
+
+        try {
+            return YAML_MAPPER.readValue(inputStream, BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in Yaml format");
+        }
+
+        log.error("Config is not in a valid format");
```

Comment:
I would add more details about the supported formats (just saying that JSON and YAML are supported)

Reply:
Also what if the file path does not exist? we do not distinguish between file not found and wrong format

Reply:
> I would add more details about the supported formats (just saying that JSON and YAML are supported)

I think it will be determined in the documentation

Reply:
Yes but as we are very much aware developers don't read the doc :) 
If we can provide helpful error message that suggest how to fix an error that will improve the developer's experience